services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/service/NodeService.java

```diff
@@ -26,4 +26,8 @@
     NodeInfo updateNodeInfo(String nodeId, NodeInfo nodeInfo) throws ParameterNullOrEmptyException, InvalidDataException, Exception;
 
     String deleteNodeInfo(String nodeId) throws ParameterNullOrEmptyException, Exception;
+ 
+    // TEMP: URI will be provided by the Admin but these are for testing purpose.
+    static String makeUpNcmId(String hostIP, int localPort) { return "ncm_" + hostIP + "_" + String.valueOf(localPort); }
```

Comment:
I realized that creating an ID is required. However, it cannot be tied to any specific node, as you noted. So, this will be makeUpNcmId will be made official but I think I will find a better name, something like create/generateNcmId.